docs/development/setup-local.md

```diff
@@ -0,0 +1,254 @@
+```sh
+# Log in to your postgres service.
+; psql
+
+# These match the defaults in the `.env.example` file.
+# You can use different values here, just make sure to modify your `.env` file accordingly.
+; create user attune with login password 'attune' superuser;
+; create database attune owner attune;
+```
+
+## Configure local object storage
+
+This services requires S3 compatible object storage,
+these instructions walk through using Minio.
+
+If running Minio natively, provide it with the same settings as you'll use in the `.env` file:
+```sh
+# These match the defaults in the `.env.example` file.
+# You can use different values here, just make sure to modify your `.env` file accordingly.
+MINIO_DOMAIN=localhost MINIO_ROOT_USER=attuneminio MINIO_ROOT_PASSWORD=attuneminio minio server ~/.minio/data
+```
+
+You can validate that minio is running with the `mc` command:
+```sh
+; mc alias set 'attune' 'http://127.0.0.1:9000' 'attuneminio' 'attuneminio'
+; mc ready attune
+```
+
+> [!WARNING]
+> Minio uses subdomains for the bucket.
+> On macOS you'll need to manually add the bucket name you configured as the subdomain:
+> ```text
+> 127.0.0.1	      localhost      attune-dev-0.localhost
+> 255.255.255.255	broadcasthost
+> ::1             localhost
+> ```
+>
+> You can validate that it is working properly by hitting its health endpoint:
+> ```sh
+> ; xh attune-dev-0.localhost:9000/minio/health/live
+> ```
+
+Finally, you may need to set the bucket as "public read only" for it to be read by `apt`:
+```sh
+; mc anonymous set download attune/attune-dev-0 -r
+```
+
+You can validate that this works with this:
+```sh
+; mc anonymous list attune/attune-dev-0 -r
+attune-dev-0/* => readonly
+```
+
+## Set up `.env` file
+
+Set up your `.env` file using the `.env.example` file as a base.
+
+Make sure to follow any instructions in the comments,
+and make sure that the values match the values you used when configuring services earlier.
+
+> [!TIP]
+> A convenient one-liner for loading the contents of the `.env` file into your current session
+> is as follows:
+> ```sh
+> set -a; source .env; set +a
+> ```
+
+## Migrate the database
+
+Migrations currently use Prisma.
+
+To run a migration manually, install Node and navigate to the `docker/migrate` folder.
+Install the dependencies and run migrations; make sure that `ATTUNE_DATABASE_URL` is set correctly.
+
+```sh
+; npm i
+; npm run migrate
+```
+
+## Run Attune
+
+Finally, you can now run the Attune server!
+
+Make sure that your `.env` file is loaded or that you specify the arguments manually,
+then run Attune with the following:
+```sh
+; cargo run attune
+```
+
+You can validate that the server is running by hitting the `/api/v0/repositories` endpoint.
+Here's an example using [`xh`](https://github.com/ducaale/xh):
+```sh
+; xh :3000/api/v0/repositories --bearer $ATTUNE_API_TOKEN
+```
+
+Keep this server running; future steps will be in a new terminal window.
+
+## Publish a package
```

Comment:
How much of this is duplicated with https://github.com/attunehq/attune/blob/main/docs/user-guide/README.md? I'd rather point one to the other than maintain two sources of truth.

Reply:
Indifferent, up to you.

Reply:
I'll merge as is for now then, when we do the upcoming doc updates I'll revisit then.